src/mapgen/treegen.cpp

```diff
@@ -44,6 +44,12 @@ void make_tree(MMVManip &vmanip, v3s16 p0, bool is_apple_tree,
 	MapNode treenode(ndef->getId("mapgen_tree"));
 	MapNode leavesnode(ndef->getId("mapgen_leaves"));
 	MapNode applenode(ndef->getId("mapgen_apple"));
+	if (treenode == CONTENT_IGNORE)
```

Comment:
does this comparison even work?
treenode is a `MapNode` (struct) and CONTENT_IGNORE is a number

Reply:
Apparently it does, when `CONTENT_IGNORE` is constructed as MapNode and then compared to `treenode`. Strange that it seems to happen automatically.

https://github.com/minetest/minetest/blob/12275e235a17e84805f4b3b667e1ddbdf9499fc6/src/mapnode.h#L142-L153